File: spring-cloud-square-demo/consumer-okhttp/src/main/java/com/livk/consumer/RemoteController.java
package com.livk.consumer;

import lombok.RequiredArgsConstructor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import org.springframework.http.HttpEntity;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.io.IOException;

/**
 * <p>
 * RemoteController
 * </p>
 *
 * @author livk
 * @date 2022/4/13
 */
@RestController
@RequiredArgsConstructor
public class RemoteController {

	private final OkHttpClient.Builder builder;

	// 本质就是使用okhttp调用，和restTemplate没什么区别
	@GetMapping("/remote/instance")
	public HttpEntity<String> remoteInstance() throws IOException {
		Request request = new Request.Builder().url("http://square-provider/instance").build();
		try (Response response = builder.build().newCall(request).execute()) {
			return ResponseEntity.ok(response.body().string());
		}
	}

}

File: spring-cloud-square/consumer-okhttp/src/main/kotlin/com/livk/consumer/RemoteController.kt
package com.livk.consumer

import okhttp3.OkHttpClient
import okhttp3.Request
import org.springframework.http.HttpEntity
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @author livk
 */
@RestController
class RemoteController(private val builder: OkHttpClient.Builder) {

    // 本质就是使用okhttp调用，和restTemplate没什么区别
    @GetMapping("/remote/instance")
    fun remoteInstance(): HttpEntity<String> {
        val request: Request = Request.Builder().url("http://square-provider/instance").build()
        return builder.build().newCall(request).execute().use { response ->
            ResponseEntity.ok(response.body!!.string())
        }
    }
}
